cloudflared/DOCS.md

```diff
@@ -144,6 +148,41 @@ additional_hosts:
 anymore. Nevertheless, you should also manually delete the DNS entry from
 Cloudflare since it can not be deleted by the Add-on._
 
+### Option: `strict_ssl`
```

Comment:
I don't think this option is needed and the description is indeed wrong:

1. I think once we decide to generate_cert, we should automatically remove the option `no-tls-verify: true`. So no need for a separate option here.
2. As I mentioned before, SSL strict can already be used in Cloudflare. The option `no-tls-verify: true` does not affect this. Strict only means, that the connection to the origin server are encrypted. Since we are using the encrypted tunnel, it seems like Cloudflare is fine with this.

Reply:
I agree in general. One further question to 1.)

I am not sure if we really can remove noTLSVerify: true after setting up Cloudflare zone certificate for e.g. homeassistant. We call homeassistant server with https://homeassistant:8123 and assign our origin certificate issued for *.example.com and example.com. I assume the TLS check will fail as hostname and certificate host will mismatch. 
I think to avoid this error we have to add another ingress rule [originServerName](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/configuration/configuration-file/ingress#originservername), and pass the Cloudflare zone name (e.g. example.com). To be honest, I have not looked at this more closely.

Reply:
You are completely right, very important point. My understanding is also be that we need to add the origin server name as another option of `originRequest` here and the check should work only after that.

Reply:
One thing to clarify here @elcajon: do you agree that we should imply strict ssl for the HA connection once the user sets `generate_cert ` and only have the option for the `additional_hosts`?

Reply:
I would agree on that. We should then also add the originRequest - originServerName automatically analogous to hostname_external for HA, right?

Reply:
`noTLSVerify: true` does not make sense to fail if the hostname mismatches the certificate. Cloudflared is meant to connect origin servers (obviously not through the domain name, but by the hostname/IP in network). So, just wondering, have someone tested this? I believe the verification that `noTLSVerify: true` is to verify whether the origin certificate matches the domain of the final domain name, not the origin domain name.

It also checks if the CA that issued the certificate is either a public valid one like Let's Encrypt or Cloudflare Origin's.

But besides that, I do agree 100% that the option should not be named `strict_ssl`. I just checked and even though the addon is talking to my Home Assistant over HTTP (not HTTPS), I could enable Strict SSL in the Cloudflare panel, and it works.

So, bottom line is that Strict SSL has absolutely nothing to do with what I proposed.

Perhaps I should change it to something like `tls_verify`, which would:

1. Be false by default
2. Be always true if "generate_cert" is enabled
3. But can also be enabled manually, in case user is using certificates generated by other means, like Let's Encrypt.

So, by this I think that the option should be kept as opposed to be removed as initially proposed by @brenner-tobias.

WDYT?

Reply:
Oh, and also, for now everything in this PR is only valid for HA. We can think about additional hostnames later. As suggested by @brenner-tobias;

Reply:
I did a quick check with my ProxmoxPVE host. After adding private key and certificate (both extracted from cert.pem) to the host and setting noTLSVerify:false the connection fails with Cloudflare SSL full (strict). 
After adding the originServerName, connection could be established. 

This is the working configuration part: 

```yaml
- hostname: proxmox.example.com
  service: https://10.10.0.2:8006
  originRequest:
    disableChunkedEncoding: true #required for Proxmox, not needed for SSL in general
    noTLSVerify: false
    originServerName: example.com
```

Reply:
So this means, that once you add the certificates, they have to be valid for the given service or you have to disable TLS verify or tell Cloudflare which originServerName to expect, right?
Thinking about the strict ssl flag, I am torn back and forth. Now talking about the HA service: I think there is no case where anyone would create a certificate another way then using the one from Cloudflare. In any case, this would not be best practice, sind we are operating in the Cloudflare Ecosystem and also the automatic renewal would not work so easily. Therefor, I still think that we should try to keep it simple and do not offer that option, unless you see any other cases I am missing.

Reply:
I think we don't need to bother about automatic renewal for now... the certificates issued by Cloudflare Origin CA are by default valid for 15 years. It can be a later improvement.

Reply:
And, generally speaking, I find that giving users more control over the underlying tools is always better, so they can extend their use cases if they want.

But I do agree that we should not bother offering every single Cloudflared configuration key. As we already allow users to set their own custom cloudflared configuration through the _Option: custom_config_, I think it may actually be a better fit to let users specify there if they want.

The only problem, however, is that the addon will always override `noTlsVerify`, thus not allowing the users to set it effectively. Am I right?

Reply:
It will not override anything in the config if the users uses a custom config. So we can completely disregard that part and only focus on the "average" user here.